tt_metal/distributed/mesh_device.hpp

```diff
@@ -169,15 +160,27 @@ class MeshDevice : public std::enable_shared_from_this<MeshDevice> {
     std::vector<std::shared_ptr<MeshDevice>> create_submeshes(
         const MeshShape& submesh_shape, MeshType type = MeshType::RowMajor);
 
-    size_t num_program_cache_entries() const;
-
     static std::shared_ptr<MeshDevice> fetch_mesh_device(const std::vector<Device*>& devices);
     static std::shared_ptr<MeshDevice> create(
         const MeshDeviceConfig& config,
         size_t l1_small_size = DEFAULT_L1_SMALL_SIZE,
         size_t trace_region_size = DEFAULT_TRACE_REGION_SIZE,
         size_t num_command_queues = 1,
         const DispatchCoreConfig& dispatch_core_config = DispatchCoreConfig{});
+
+    // Device API Queries (API contract with Device class to be supported in future)
+    CoreCoord compute_with_storage_grid_size() const;
```

Comment:
I wonder if we can start our `concept DeviceLike` in this PR, since we are here :)

Reply:
yes, as we discussed. I'll just static assert on the common methods between Device/MeshDevice for now.